proposed/2022/NuGetExeWarnWhenSDKProjectsGetPacked.md

```diff
@@ -0,0 +1,97 @@
+
+#### How packing and SDK-based project with NuGet.exe behaves today
+
+Scenario 1: Single target framework SDK based projects, with .NET SDK installed on the machine.
+
+```console
+Attempting to build package from 'SDK.csproj'.
+MSBuild auto-detection: using msbuild version '17.4.0.37601' from 'C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\Bin\amd64'.
+Packing files from 'C:\Code\Temp\SDK\bin\Debug\net6.0'.
+WARNING: NU5115: Description was not specified. Using 'Description'.
+WARNING: NU5115: Author was not specified. Using 'Roki2'.
+WARNING: NU5128: Some target frameworks declared in the dependencies group of the nuspec and the lib/ref folder do not have exact matches in the other location. Consult the list of actions below:
+- Add a dependency group for net6.0 to the nuspec
+Successfully created package 'C:\Code\Temp\SDK\SDK.1.0.0.nupkg'.
+```
+
+The command packages the single assembly, but does not generate the correct dependencies or any other metadata in the same way that the pack SDK would.
+
+Scenario 2: Multiple target framework SDK based projects, with .NET SDK installed on the machine.
+
+```console
+Attempting to build package from 'SDK.csproj'.
+MSBuild auto-detection: using msbuild version '17.4.0.37601' from 'C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\Bin\amd64'.
+Error NU5012: Unable to find 'bin\Debug\SDK\bin\Debug\'. Make sure the project has been built.
+```
+
+NuGet.exe does not understand the concept of multitargetting and just merges the output paths.
+
+Scenario 3: SDK-based project, without the .NET SDK installed
+
+```console
+The SDK 'Microsoft.NET.Sdk' specified could not be found.  C:\Code\Temp\SDK\SDK.csproj
+```
+
+The parsing fails early.
+
+### Proposal
+
+When someone attempts to run pack on an SDK-based project, NuGet will fail with the following error:
+
+```console
+Error NU5049: NuGet.exe does not support packing SDK-based projects, use `dotnet pack` to pack this project instead.
+```
+
+### Technical explanation
+
+As described above, whenever there's no SDK installed, there's simply no accurate way for NuGet to detect that a project is SDK-based.
+We can do some XML parsing, but it is likely to be error prone. `NuGet.exe pack` is not the only command that will fail, so the customer will likely figure out that they need to install the .NET SDK. After that the same scenarios with the installed .NET SDK will follow.
+
+To determine whether a project is `.NET SDK` based, we will evaluate the project (part of the normal pack operation), and check for the `UsingMicrosoftNETSDK` property being set. If the property is set, then the project is presumed to be .NET SDK based and will be blocked from packing.
+
+## Drawbacks
+
+- The proposal here would be `breaking` for customers, but it's more than likely that the scenario for which they used nuget.exe pack on an SDK based project never worked correctly.
+
+## Rationale and alternatives

```

Comment:
Thanks for listing out the alternatives. I think this approach makes plenty sense to help people use the right tool for the job.

I don't like doing nothing, because it doesn't help our problem at all.

I don't see full pack SDK support being something we can do nor would want to do for the LTS. It is an opportunity if it's the final thing we may need to add to NuGet.exe though, so not off the table.

The warning scenario is interesting though. I just worry people will absolutely ignore it and just continue doing what they've been doing though.

The disruption could be seen as potentially impactful to various CI platforms using "nuget" pre-built actions. The scarier thought is thinking how many people are using nuget to pack incorrectly in the first place for SDK-style scenarios. I think we ought to correct the behavior and bite the bullet with some blog messaging prior to release? That way people have some time to know it's coming.

Reply:
I think announcing it/blogging preemptively if we add a warning is *a great idea*. 

Given the timeline, I think targeting 6.5 would give us plenty of time to post about that. 
Very difficult to know if we'd reach the right audience though :/

Reply:
We'll blog/announce the breaking change.